components/dashboard/ActionStatusExport.js

```diff
@@ -9,193 +7,18 @@ import {
-    let ontimeTasks = 0;
-    let lateTasks = 0;
-    let completedTasks = 0;
-    const nowDate = new Date();
-
-    tasks.forEach((task) => {
-      const taskDue = new Date(task.dueAt);
-      switch (task.state) {
-        case 'NOT_STARTED':
-        case 'IN_PROGRESS':
-          if (taskDue < nowDate) lateTasks += 1;
-          else ontimeTasks += 1;
-          break;
-        case 'COMPLETED':
-          completedTasks += 1;
-          break;
-        default:
-          tasksCount -= 1;
-      }
-    });
-
-    const getOrgName = ({ organization }) => organization.name;
-
-    const parties = act.responsibleParties;
-    const primaryResponsibleOrgs = parties
-      .filter((p) => p.role === 'PRIMARY')
-      .map(getOrgName);
-    const collaboratorResponsibleOrgs = parties
-      .filter((p) => p.role === 'COLLABORATOR')
-      .map(getOrgName);
-    const otherResponsibleOrgs = parties
-      .filter((p) => p.role === null)
-      .map(getOrgName);
-
-    const row = [
-      act.identifier,
-      actionName,
-      status?.name,
-      activePhaseName,
-      new Date(act.updatedAt),
-      ontimeTasks,
-      lateTasks,
-      completedTasks,
-      tasksCount,
-      primaryResponsibleOrgs.join(';'),
-      collaboratorResponsibleOrgs.join(';'),
-      otherResponsibleOrgs.join(';'),
-    ];
-    if (multiplePlans) {
-      row.unshift(act.plan.name);
-    }
-    worksheet.addRow(row);
-  });
-
-  const today = new Date().toISOString().split('T')[0];
-  switch (fileFormat) {
-    case 'excel':
-      const xls64 = await workbook.xlsx.writeBuffer({ base64: true });
-      fileSaver.saveAs(
-        new Blob([xls64], {
-          type: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
-        }),
-        `${t('actions', getActionTermContext(plan))}-${today}.xlsx`
-      );
-      break;
-
-    case 'csv':
-      const csv64 = await workbook.csv.writeBuffer({ base64: true });
-      fileSaver.saveAs(
-        new Blob([csv64], { type: 'text/csv' }),
-        `${t('actions', getActionTermContext(plan))}-${today}.csv`
-      );
-      break;
-
-    default:
-      throw new Error('Unknown file format');
-  }
-}
-
 export default function ActionStatusExport({ actions }) {
   const t = useTranslations();
   const plan = usePlan();
-  const { actionStatuses } = plan;
-  const handleExport = async (format) => {
-    await exportActions(t, actions, actionStatuses, plan, format);
-  };
+  const url = plan.actionReportExportViewUrl;
+  const csvExportUrl = `${url}?format=csv`;
+  const excelExportUrl = `${url}?format=xlsx`;
   return (
     <UncontrolledDropdown>
       <DropdownToggle caret>{t('export')}</DropdownToggle>
       <DropdownMenu>
-        <DropdownItem onClick={() => handleExport('excel')}>Excel</DropdownItem>
-        <DropdownItem onClick={() => handleExport('csv')}>CSV</DropdownItem>
+        <DropdownItem href={excelExportUrl}>Excel</DropdownItem>
```

Comment:
Is `href` specifically supported by `DropdownItem`? If not, we should override the underlying element be an `<a>` using `tag="a"` (I think that's the prop).

You probably also want to use [the `download` property](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#download) here.

Reply:
export interface DropdownItemProps extends React.HTMLAttributes<HTMLElement> {
  [key: string]: any;
  active?: boolean;
  disabled?: boolean;
  divider?: boolean;
  tag?: React.ElementType;
  header?: boolean;
  cssModule?: CSSModule;
  href?: string;
  toggle?: boolean;
  text?: boolean;
}
I think href is supported!

Reply:
```      } else if (props.href) {
        Tag = 'a';
      } else if (text) {
```

Reply:
Great! Well found, they don't mention it [in the docs](https://reactstrap.github.io/?path=/docs/components-dropdown--dropdown)

Reply:
If we want to be on the safe side, we could continue using `file-saver` but simplify that original `handleExport` to just download the specified URL.

```ts
  import { saveAs } from 'file-saver';
  ...
  const handleExport = async (downloadUrl) => {
    saveAs(downloadUrl);
  };
```

I've not tested that, but it's probably safer than letting browsers decide whether to allow our cross-origin download (I should do the same for NetZeroPlanner)